speedtest/latency_test.go

```diff
@@ -0,0 +1,68 @@
+package speedtest
+
+import (
+	"errors"
+	"io"
+	"net/http"
+	"testing"
+	"time"
+)
+
+func Test_measureLatency(t *testing.T) {
+	tests := []struct {
+		name   string
+		client Client
+		input  time.Duration
+		want   time.Duration
+	}{
+		{
+			name:   "Client.Get() error with DefaultErrorLatency input",
+			client: &latencyErrorClient{},
+			input:  DefaultErrorLatency,
+			want:   DefaultErrorLatency,
+		},
+		{
+			name:   "Client.Get() error with 10 Second input",
+			client: &latencyErrorClient{},
+			input:  10 * time.Second,
+			want:   10 * time.Second,
+		},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			s := &Server{client: tc.client}
```

Comment:
This is possible, to set the different client, because of the `type Client interface {..}` change, as I mentioned in the previous comment.